### PR TITLE
Return sample count in label API responses

### PR DESCRIPTION
**Pull request by tianshengsui**
ch-102

### CODE REVIEW

File: virtool/labels/db.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+async def attach_sample_count(db, document, label_id):
+    document.update({"count": await db.samples.count_documents({"labels": {"$in": [label_id]}})})
```
Comment (igboyes) [inline virtool/labels/db.py:2]: This should work:
```py
await db.samples.count_documents({"labels": label_id})
```

Right now this function relies on a side-effect to work: it modifies `document` from the outside scope. A purer approach is to return a new `dict`:

```py
async def attach_sample_count(db, document, label_id):
  return {
    **document,
    "count": await db.samples.count_documents({"labels": {"$in": [label_id]}})
  }
```

File: virtool/labels/db.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+async def attach_sample_count(db, document, label_id):
```
Comment (igboyes) [inline virtool/labels/db.py:1]: Needs docstring and typing.

Write specific tests for this function instead of relying on the integration tests for the API endpoints. In the API endpoint tests, you only test the case where each sample has one label. Add unit tests for this function that work with cases where a sample has zero or more labels.

**Review by igboyes — CHANGES_REQUESTED** [review]

**Review by igboyes — APPROVED** [review]
Great. Thanks!
